core/src/main/java/com/graphhopper/util/StopWatch.java

```diff
@@ -81,6 +81,13 @@ public long getMillis() {
         return elapsedNanos / 1_000_000;
     }
 
+    /**
+     * returns the elapsed time in ms but includes the fraction as well to get a precise value
+     */
+    public double getMillisWithFraction() {
+        return elapsedNanos / 1_000_000.0;
+    }
+
```

Comment:
Now we have `getSeconds` which returns a float, `getMillis` and `getNanos` which return a long and `getMillisWithFraction` which returns a double :D As far as I am concerned we could just keep `long getNanos()`, add `long getCurrentNanos()` and delete the rest so whoever needs to format this to some other unit can do this with some static helper methods. 

Then again, I think it's a good choice to return `double` in `getMillisWithFraction`, rather than `float`. Why does `getSeconds()` return a float anyway? Maybe rename to `getMillisDouble`? I don't really like the `withFraction` thing, because it sounds like something special is going on when all it does is return a floating point number.

But anyway, these kind of things can be discussed forever with rather minor benefits of the many different ways to do/name things :)

Reply:
> Why does getSeconds() return a float anyway?

Apparently that was me btw :)

Reply:
> 👍 or better as before doing the conversion in the code directly.

I renamed it to getMillisDouble. I don't really like to do the conversion in the code, as I t hink it is less readable and more error prone. If I read `nanos / 1e6` I first need to see if this results in ms or some other unit. Well I know it now, but I am likely to forget and will have to do calculations again in the future :).

Reply:
>  I think it is less readable and more error prone.

We could use static helper methods for such conversions.

Let's just merge here :)